conngen/conngenmodule.cpp

```diff
@@ -378,14 +366,24 @@ ConnectionGeneratorModule::CGNext_cgFunction::execute( SLIInterpreter* i ) const
   ConnectionGeneratorDatum cgd =
     getValue< ConnectionGeneratorDatum >( i->OStack.pick( 0 ) );
 
-  int j, k;
-  std::vector< double > values;
+  int arity = cgd->arity();
+  double* tmp_values = new double[ arity ];
+  std::vector< double > values( arity );
+
   i->OStack.pop( 1 );
-  i->EStack.pop();
-  if ( cg_next( cgd, j, k, values ) )
+
+  int source_id, target_id;
+  if ( cgd->next( source_id, target_id, tmp_values ) )
   {
-    i->OStack.push( j );
-    i->OStack.push( k );
+    for ( int m = 0; m < arity; ++m )
+    {
+      values[ m ] = tmp_values[ m ];
+    }
+    delete[] tmp_values;
```

Comment:
This delete only occurs in this branch of the if statement

Reply:
`cgd->next()` won't put any more values into `tmp_values` before it returns `false`. So we don't have to delete anything in the `else` branch.